pkg/kubelet/cm/dra/manager.go

```diff
@@ -56,6 +56,31 @@ func NewManagerImpl(kubeClient clientset.Interface) (*ManagerImpl, error) {
 	return manager, nil
 }
 
+// Admit checks if pod is in the list of users for the claim
+func (m *ManagerImpl) Admit(pod *v1.Pod) error {
+	for _, podResourceClaim := range pod.Spec.ResourceClaims {
+		claimName := resourceclaim.Name(pod, &podResourceClaim)
+		klog.V(3).InfoS("Check if pod can be admited", "pod", pod.Name, "claim", claimName)
+		// Query claim object from the API server
+		resourceClaim, err := m.kubeClient.CoreV1().ResourceClaims(pod.Namespace).Get(context.TODO(), claimName, metav1.GetOptions{})
```

Comment:
The same claim will be needed again later, i.e. it will now be fetched twice?

Can this be avoided, perhaps by doing this check at the time when the claim is about to be used?

When we discussed it, I said that I consider this a final error. But I don't feel strongly about this because one could also argue that it is an oversight that can be fixed by manually adding the pod to `ReservedFor` after pod scheduling.

So if the implementation becomes simpler when doing the check later, even when that means enabling retrying, then we probably should do it. 

kubelet should not become less efficient because of a situation that should never occur.

Reply:
OK, I'll close this PR and enable another one that checks `ReservedFor` just before calling PrepareResource.